chat/src/main/java/com/github/twitch4j/chat/TwitchChat.java

```diff
@@ -451,20 +496,37 @@ protected void onConnected() {
             boolean sendRealPass = sendCredentialToThirdPartyHost // check whether this security feature has been overridden
                 || baseUrl.equalsIgnoreCase(TWITCH_WEB_SOCKET_SERVER) // check whether the url is exactly the official one
                 || baseUrl.equalsIgnoreCase(TWITCH_WEB_SOCKET_SERVER.substring(0, TWITCH_WEB_SOCKET_SERVER.length() - 4)); // check whether the url matches without the port
-            sendTextToWebSocket(String.format("pass oauth:%s", sendRealPass ? chatCredential.getAccessToken() : CryptoUtils.generateNonce(30)), true);
-            userName = String.valueOf(chatCredential.getUserName()).toLowerCase();
+
+            String token;
+            if (sendRealPass) {
+                if (validateOnConnect && connection.getConfig().backoffStrategy().getFailures() > 1 && !identityProvider.getAdditionalCredentialInformation(chatCredential).isPresent()) {
```

Comment:
we need to call validate in a manner that allows us to distinguish network failure from expired token; currently, if there's a network issue, once it alleviates, we may end up reconnecting in anonymous mode, when the token wasn't ever invalid